hl7/4.2/modules/ROOT/edits/index.adoc

```diff
@@ -0,0 +1,174 @@
+|Errors (read only) |A list of errors associated with the input message.
+|Id |Message structure ID
+|MSH (read only) |Link that receives MSH segment data
+|Name (read only) |Message structure name
+|===
+
+Individual messages have their own maps, with keys matching the segments of the message. For example, an ACK message uses the message structure ID ACK, and the data for the sent or received ACK message is an ACK value in the Data map. The ACK message is itself a map, and the segments and groups of the message are represented as maps (in the case of singleton instances) or lists of maps (for repeating instances) with positional keys.
+
+There are two special cases where using generic handling for data that is not included in a schema definition.
+
+The first is for HL7 values of the `varies` type. Since these values consist of any structure of components and subcomponents that can be repeated, the parser uses a list of map representations for each `varies` of type. The keys in each map are generated as the value is parsed, matching standard HL7 value names with two digits used for each nesting level.
+So a simple text value for an `OBX-05 Observation Value` field, would use the key `OBX-05` in a map. If there are two components present, they use keys `OBX-05-01` and `OBX-05-02`.
+
+The second is extension segments with tags that match a pattern configured under parser options. These are similar in structure to the `varies` values, with the exception of only being within a single map for the entire segment.
+
+The maps containing extension segment data are added to the basic message map in lists with the key `ExtensionSegs`. In addition to the actual extension segment data, the map for the extension segment contains two other keys:
+
+[%header%autowidth.spread]
+|===
+|Key |Description
+|Ident |The extension segment identifier (tag)
+|Position |The position of the segment within the message structure, as a two-digit string. This is the same as the position of the immediately preceding defined segment, as defined in the schema. If a `ZVN` extension segment is used following the `EVN` segment in an `ADT_A01` message structure, the `ZVN` is at position `03)``.
+|===
+
+If extension segments are used in nested groups, the list containing those segments are included in the map representing that group. Extension segments are ordered by position in the lists created by the parser, and must also be ordered by position when writing.
+
+=== About Sending Acknowledgments
+
+An ACK (acknowledgment) message is an HL7 message that lets you acknowledge to a message sender that your application has received a message. ACK messages are the same as any other HL7 message writes, except that you set the ACK message to what was generated during the read operation as the output message under a `Data` key.
+
+For example:
```

Comment:
GLOBALLY replace this fragment introduction with a complete sentence, per style ref.